### PR TITLE
Added Filter validation for google_scc_notification_config

### PR DESCRIPTION
**Pull request by thokalavinod**
TestAccSecurityCenterNotificationConfig_updateStreamingConfigFilter 
 api: 'https://cloud.google.com/security-command-center/docs/reference/rest/v1/organizations.notificationConfigs'

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
securitycenter: empty `streaming_config.filter` values are now supported in `google_scc_notification_config` resources
```

### CODE REVIEW

**Review by NickElliot — APPROVED** [review]
LGTM -- going to verify an upgrade scenario locally then merge through if all looks good!

File: mmv1/products/securitycenter/NotificationConfig.yaml [hunk 1]
```diff
@@ -94,6 +96,9 @@ properties:
       - !ruby/object:Api::Type::String
         name: filter
         required: true
+        send_empty_value: true
+        validation: !ruby/object:Provider::Terraform::Validation
+          function: 'validation.StringLenBetween(0,1024)'
```
Comment (NickElliot) [inline mmv1/products/securitycenter/NotificationConfig.yaml:101]:
```suggestion
```

Adding this new validation can theoretically break users -- we should let it be handled by the API

**Review by NickElliot — CHANGES_REQUESTED** [review]

**Comment by NickElliot** [comment]
one change needed

File: mmv1/products/securitycenter/NotificationConfig.yaml (hunk 1 above)
Comment (thokalavinod) [reply mmv1/products/securitycenter/NotificationConfig.yaml]: Addressed

**Comment by thokalavinod** [comment]
> one change needed

fixed

**Review by NickElliot — APPROVED** [review]
LGTM

**Pull request merged**